compiler+runtime/test/cpp/jank/runtime/obj/persistent_string.cpp

```diff
@@ -0,0 +1,70 @@
+/* clang-format off */
+#include <jank/runtime/obj/persistent_string.hpp>
+#include <jank/runtime/obj/character.hpp>
+#include <jank/runtime/obj/number.hpp>
+#include <jank/runtime/obj/nil.hpp>
+#include <jank/runtime/core.hpp>
+
+/* This must go last; doctest and glog both define CHECK and family. */
+#include <doctest/doctest.h>
+
+namespace jank
+{
+  TEST_SUITE("persistent_string")
+  {
+    TEST_CASE("Index")
```

Comment:
We're missing test cases for the `throw` in each `get`, where the input is not an integer.

Reply:
`get` does not throw.

Reply:
But there's definitely a bug in the `get` fallback, I need another case.

Reply:
Oh, whoops. I meant `contains` throwing.

Reply:
What do you think about returning `false` instead of throwing for `(contains? "a" non-int)`? Clojure throws, CLJS returns false.

Reply:
If CLJS sets a precedent for `false`, I'd follow that. Exceptions for predicates are bonkers.

Reply:
Changed `(contains? "a" non-int)` to return false and added your suggested test cases and a couple more.